### PR TITLE
Add file sorting

### PR DESCRIPTION
**Pull request by Kr0nox**
This PR adds a selector at the top of the comparison view to the top of the comparison view.
grafik
 Files can be sorted
- Alphabetically
- By the percent of tokens that are part of a match
- The number of matches in a file
- The size of the matches in the files

### CODE REVIEW

File: report-viewer/src/components/fileDisplaying/FilesContainer.vue [hunk 1]
```diff
@@ -91,7 +91,78 @@ const props = defineProps({
...
+  }
+  return matches
+})
+
+const sortedFiles: Ref<SubmissionFile[]> = ref([])
+sortFiles(store().uiState.fileSorting ?? FileSortingOptions.ALPHABETICAL)
+
+function sortFiles(fileSorting: FileSortingOptions) {
+  switch (fileSorting) {
+    case FileSortingOptions.ALPHABETICAL: {
```
[the remark this answers is not in the document]
Comment (Kr0nox) [reply report-viewer/src/components/fileDisplaying/FilesContainer.vue:111]: No, that is not possible with enums in typescript

File: report-viewer/src/model/ui/FileSortingOptions.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    tooltip: 'Sort files by the percentage of tokens that are part of a match.'
+  },
+  [FileSortingOptions.MATCH_COUNT]: {
+    displayValue: 'Match Count',
+    tooltip: 'Sort files by the number of matches found in the file.'
+  },
+  [FileSortingOptions.MATCH_SIZE]: {
+    displayValue: 'Match Size',
+    tooltip: 'Sort files, so that the largest matches are at the top.'
+  }
```
Comment (tsaglam) [inline report-viewer/src/model/ui/FileSortingOptions.ts:26]: Minor phrasing changes:
```suggestion
  [FileSortingOptions.ALPHABETICAL]: {
    displayValue: 'Alphabetical',
    tooltip: 'Sort files alphabetically, similar to package structure.'
  },
  [FileSortingOptions.MATCH_COVERAGE]: {
    displayValue: 'Match Coverage',
    tooltip: 'Sort files by the percentage of tokens included in a match.'
  },
  [FileSortingOptions.MATCH_COUNT]: {
    displayValue: 'Match Count',
    tooltip: 'Sort files by the number of matches found.'
  },
  [FileSortingOptions.MATCH_SIZE]: {
    displayValue: 'Match Size',
    tooltip: 'Sort files by match size, with the largest matches at the top.'
  }
```

**Review by tsaglam — APPROVED** [review]
Just a comment regarding tooltips, but besides that it is ready to merge.

**Comment by tsaglam** [comment]
@TwoOfTwelve ready from your side?

**Pull request merged**